src/curl.c

```diff
@@ -123,13 +123,14 @@ static CURLcode swupd_curl_set_timeouts(CURL *curl)
 	return curl_ret;
 }
 
-static int check_connection(const char *test_capath)
+static int check_connection(const char *test_capath, char *url)
 {
 	CURLcode curl_ret;
 	long response = 0;
+	char *connection_url = url ? url : version_url;
```

Comment:
It's a good idea to try to remove dependencies from swupd.h in this file. So take the version_url out. We can always use url. If url == NULL returns an error.

Reply:
good idea